source/validate_decorations.cpp

```diff
@@ -160,9 +166,13 @@ uint32_t getSize(uint32_t member_id, bool roundUp, ValidationState_t& vstate) {
       const auto componentSize = getSize(componentId, roundUp, vstate);
       return componentSize * numComponents;
     }
-    case SpvOpTypeArray:
-      return (vstate.FindDef(words[3])->words()[3] - 1) * baseAlignment +
+    case SpvOpTypeArray: {
+      const auto sizeInst = vstate.FindDef(words[3]);
+      if (SpvOpSpecConstant == sizeInst->opcode()) return 0;
```

Comment:
Sorry I wasn't clearer when I said OpSpecConstant. I meant the whole family. Use `spvOpcodeIsSpecConstant()`.